app/models/model_set.rb

```diff
@@ -50,6 +50,7 @@ def collection_to_delete
     # Remove all elements to be deleted from collection and return them
     # Allows us to render @model_set.collection without deleted elements
     deleted = []
+    @collection = collection.to_a
     collection.delete_if { |e| deleted << e if @delete_if.andand.call(e.attributes) }
```

Comment:
Isn't `collection` referencing the same object as `@collection` here? Can we just do `collection.to_a.delete_if...` without the extra assignment in line 53?

Reply:
I dont think so, we need to convert the attribute to_a, if we do what you suggest the delete_if will only work on the converted array, not the attribute.

Reply:
I got bitten by this a few weeks ago (I couldn't find the PR but Luis and I had a similar conversation). I was passing `collection.to_a` into a method and the method was operating on the converted array. So it makes sense that we need the extra line.